security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -533,7 +566,7 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName s
 	}
 
 	// Cert exipre time by default is createTime + sc.configOptions.SecretTTL.
-	// Current CAs(Citadel, GoogleCA) respect SecretTTL that passed to it and use it decide TTL of cert it issued.
+	// Current CAs(Citadel, etc) respect SecretTTL that passed to it and use it decide TTL of cert it issued.
```

Comment:
The generateSecret function has more than 100 lines. Consider breaking it into small pieces for readability concern. Can be in a separate PR

Reply:
resolve other comments, will do this one in separate PR